### PR TITLE
Refactor sign-up error messages

### PR DESCRIPTION
**Pull request by arthurashman**
We used `user.errors.full_messages` which gives you an array of
human-readable error messages, which was so simple we left
it in the controller and didn't refactor it into a helper after all.

Plus we refactored the posts index view to get the posts's author's
email address directly and deleted the helper method. Neater, and we
delegated the `.email` method from the post through to the post's
author in accordance with the [Law of Demeter](https://github.com/Hives/instagram-challenge/blob/master/docs/review.md#law-of-demeter-violations).

- Paul

### CODE REVIEW

**Comment by Hives** [comment]
[Refactor some of the stuff from UserController into helper methods [SMALL]](https://trello.com/c/l9t5Mlgd/33-refactor-some-of-the-stuff-from-usercontroller-into-helper-methods-small)

**Review by JRRS1982 — COMMENTED** [review]
Looks fine to me :o)

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -11,9 +11,11 @@
   <div class='post-card'>
     <div class="post-header">
       <div class="post-author">
-        <%= get_user_email(post) %>
+        <%= post.email %>
+      </div>
+      <div class="post-pubished-time">
+        <%= post.created_at.strftime("%d-%m-%Y")%> at <%= post.created_at.strftime("%I:%M %p")%>
```
Comment (j-dav) [inline app/views/posts/index.html.erb:17]: pubished = published 

SPELLING!

**Comment by j-dav** [comment]
Looks great, law of demeter interesting. Rails delegate method new to me!

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (Hives) [reply app/views/posts/index.html.erb:17]: the horror

**Comment by Hives** [comment]
i've updated the typo. apart from the small change we've got approval from jeremy and james

**Pull request merged**